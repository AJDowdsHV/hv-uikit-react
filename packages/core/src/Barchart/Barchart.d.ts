import { HvChartProps } from "../Chart";

export interface HvBarchartProps extends HvChartProps {
  /**
   * Sets is the chart is stack.
   */
  stack: boolean;
  /**
   * Sets is the chart is horizontal.
   */
  horizontal: boolean;
}

export default function HvBarchart(props: HvBarchartProps): JSX.Element | null;
