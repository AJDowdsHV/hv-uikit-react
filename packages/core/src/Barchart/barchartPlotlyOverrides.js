import clone from "lodash/cloneDeep";

const setterIfNil = (object, property, value) => {
  // eslint-disable-next-line no-param-reassign
  object[property] = object[property] || value;
};

/**
 * Auxiliary functions to set the layout in order to follow the DS guideline.
 */
export const setLayout = (inputLayout, stack) => {
  const layout = inputLayout === undefined ? {} : clone(inputLayout);
  setterIfNil(layout, "bargap", 0.25);
  setterIfNil(layout, "bargroupgap", 0.25);
  if (stack) setterIfNil(layout, "barmode", "stack");

  return layout;
};

export const setData = (inputData, isHorizontal) => {
  const data = inputData;

  data.forEach(trace => {
    setterIfNil(trace, "type", "bar");
    setterIfNil(trace, "hoverinfo", "none");
    if (isHorizontal) setterIfNil(trace, "orientation", "h");
  });

  return data;
};
