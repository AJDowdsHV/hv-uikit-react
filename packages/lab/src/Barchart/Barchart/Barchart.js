/*
 * Copyright 2019 Hitachi Vantara Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { memo } from "react";
import PropTypes from "prop-types";
import Typography from "@hv/uikit-react-core/dist/Typography";
import Plotly from "plotly.js-basic-dist";
import createPlotlyComponent from "react-plotly.js/factory";
import isNil from "lodash/isNil";
import styleCreator from "./styles";

/**
 * Setter of default properties.
 *
 * @param layout
 * @param theme
 * @returns {*}
 */
const propsSetter = (layout, theme) => {
  const styles = styleCreator(theme);
  const newLayout = layout;

  newLayout.colorway = Object.values(styles.defaultColors);

  if (isNil(newLayout.legend)) newLayout.legend = {};

  newLayout.legend.font = {
    family: styles.vizText.fontFamily,
    size: styles.vizText.fontSize,
    color: styles.vizText.color
  };

  newLayout.plot_bgcolor = theme.hv.palette.atmosphere.atmo1;

  newLayout.paper_bgcolor = theme.hv.palette.atmosphere.atmo1;

  if (isNil(newLayout.xaxis)) newLayout.xaxis = {};
  newLayout.xaxis.linecolor = styles.lineColor.color;

  if (isNil(newLayout.xaxis.title)) newLayout.xaxis.title = {};
  newLayout.xaxis.title.font = {
    family: styles.vizText.fontFamily,
    size: styles.vizText.fontSize,
    color: styles.vizText.color
  };

  newLayout.xaxis.tickcolor = styles.lineColor.color;
  newLayout.xaxis.gridcolor = theme.hv.palette.atmosphere.atmo5;
  newLayout.xaxis.tickfont = {
    family: styles.vizText.fontFamily,
    size: styles.vizText.fontSize,
    color: styles.vizText.color
  };

  if (isNil(newLayout.yaxis)) newLayout.yaxis = {};
  newLayout.yaxis.linecolor = styles.lineColor.color;
  newLayout.yaxis.gridcolor = theme.hv.palette.atmosphere.atmo5;

  if (isNil(newLayout.yaxis.title)) newLayout.yaxis.title = {};

  newLayout.yaxis.title.font = {
    family: styles.vizText.fontFamily,
    size: styles.vizText.fontSize,
    color: styles.vizText.color
  };
  newLayout.yaxis.tickcolor = styles.lineColor.color;
  newLayout.yaxis.tickfont = {
    family: styles.vizText.fontFamily,
    size: styles.vizText.fontSize,
    color: styles.vizText.color
  };

  return newLayout;
};

/**
 * Ploty barchart.
 *
 * @param classes
 * @param theme
 * @param title
 * @param subtitle
 * @param data
 * @param layout
 * @param onHover
 * @param onUnHover
 * @param onMouseMove
 * @returns {*}
 * @constructor
 */
const BarChart = ({
  classes,
  theme,
  title,
  subtitle,
  data,
  layout,
  config,
  onHover,
  onUnHover,
  onMouseMove
}) => {
  const Plot = createPlotlyComponent(Plotly);
  const newLayout = propsSetter(layout, theme);

  return (
    <div className={classes.root} onMouseMove={e => onMouseMove(e)}>
      <div className={classes.titleContainer}>
        {title && <Typography variant="mTitle">{title}</Typography>}
        <div className={classes.subtitle}>
          {subtitle && <Typography variant="infoText">{subtitle}</Typography>}
        </div>
      </div>
      <Plot
        data={data}
        layout={newLayout}
        config={config}
        onHover={eventData => onHover(eventData)}
        onUnhover={eventData => onUnHover(eventData)}
      />
    </div>
  );
};

BarChart.propTypes = {
  /**
   * A Jss Object used to override or extend the styles applied.
   */
  classes: PropTypes.shape({
    /**
     * Styles applied to the barchart root class.
     */
    root: PropTypes.string
  }).isRequired,
  /**
   * Title of the chart.
   */
  title: PropTypes.string,
  /**
   * Subtitle of the chart.
   */
  subtitle: PropTypes.string,
  /**
   * Plotly data object.
   */
  data: PropTypes.arrayOf(PropTypes.instanceOf(Object)).isRequired,
  /**
   * Plotly layout object.
   */
  layout: PropTypes.instanceOf(Object).isRequired,
  /**
   * Plotly config object.
   */
  config: PropTypes.instanceOf(Object).isRequired,
  /**
   * Theme.
   */
  theme: PropTypes.instanceOf(Object).isRequired,
  /**
   * OnHover function.
   */
  onHover: PropTypes.func,
  /**
   * OnUnHover function.
   */
  onUnHover: PropTypes.func,
  /**
   * OnMouseMove function.
   */
  onMouseMove: PropTypes.func
};

BarChart.defaultProps = {
  title: "",
  subtitle: "",
  onHover: null,
  onUnHover: null,
  onMouseMove: null
};

const arePropsEqual = (prevProps, nextProps) =>
  prevProps.title === nextProps.title &&
  prevProps.subtitle === nextProps.subtitle &&
  prevProps.data === nextProps.data &&
  prevProps.layout === nextProps.layout;

export default memo(BarChart, arePropsEqual);
