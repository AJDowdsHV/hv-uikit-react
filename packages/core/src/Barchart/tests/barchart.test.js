/* eslint-env jest */

import React from "react";
import { mount, shallow } from "enzyme";

import HvProvider from "../../Provider";
import Barchart from "..";
import { applyDataDefaults, applyLayoutDefaults } from "../barchartPlotlyOverrides";

describe("Barchart withStyles", () => {
  let wrapper;

  const trace1 = {
    x: [2300, 1000, 8500],
    y: ["January", "February", "March"],
    name: "Sales Target"
  };

  const data = [trace1];
  const layout = {};

  beforeEach(() => {
    wrapper = shallow(
      <HvProvider>
        <Barchart data={data} layout={layout} />
      </HvProvider>
    );
  });

  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(wrapper.find(Barchart)).toMatchSnapshot();
  });

  it("should render the Barchart", () => {
    wrapper = mount(
      <HvProvider>
        <Barchart data={data} layout={layout} />
      </HvProvider>
    );
    const barchart = wrapper.find(Barchart);
    expect(barchart.length).toBe(1);
  });

  it("should set default data", () => {
    const overwrittenData = applyDataDefaults(data);
    expect(overwrittenData[0].type).toBe("bar");
    expect(overwrittenData[0].hoverinfo).toBe("none");
  });

  it("shouldn't override existing data", () => {
    const traceTest = { type: "xx" };
    const overwrittenData = applyDataDefaults([traceTest]);
    expect(overwrittenData[0].type).toBe("xx");
    expect(overwrittenData[0].hoverinfo).toBe("none");
  });

  it("should set default layout", () => {
    const overwrittenLayout = applyLayoutDefaults(layout);
    expect(overwrittenLayout.bargap).toBe(0.25);
    expect(overwrittenLayout.bargroupgap).toBe(0.25);
  });
});
