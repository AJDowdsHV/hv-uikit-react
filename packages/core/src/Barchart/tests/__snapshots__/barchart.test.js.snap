// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Barchart should render correctly 1`] = `
<WithStyles(Barchart)
  data={
    Array [
      Object {
        "hoverinfo": "none",
        "name": "Sales Target",
        "type": "bar",
        "x": Array [
          "January",
          "February",
          "March",
        ],
        "y": Array [
          2300,
          1000,
          8500,
        ],
      },
    ]
  }
  subtitle="Sales performance (YTD)"
  title="Simple Vertical Bar Chart"
  xAxisTitle="Thousands of Dollars ($)"
  yAxisTitle="Axis description"
>
  <Barchart
    classes={Object {}}
    data={
      Array [
        Object {
          "hoverinfo": "none",
          "name": "Sales Target",
          "type": "bar",
          "x": Array [
            "January",
            "February",
            "March",
          ],
          "y": Array [
            2300,
            1000,
            8500,
          ],
        },
      ]
    }
    subtitle="Sales performance (YTD)"
    title="Simple Vertical Bar Chart"
    xAxisTitle="Thousands of Dollars ($)"
    yAxisTitle="Axis description"
  >
    <WithStyles(Chart)
      afterPlot={[Function]}
      classes={Object {}}
      data={
        Array [
          Object {
            "hoverinfo": "none",
            "name": "Sales Target",
            "type": "bar",
            "x": Array [
              "January",
              "February",
              "March",
            ],
            "y": Array [
              2300,
              1000,
              8500,
            ],
          },
        ]
      }
      layout={
        Object {
          "bargap": 0.25,
          "bargroupgap": 0.25,
        }
      }
      subtitle="Sales performance (YTD)"
      title="Simple Vertical Bar Chart"
      tooltipType="multiple"
      xAxisTitle="Thousands of Dollars ($)"
      yAxisTitle="Axis description"
    >
      <Chart
        afterPlot={[Function]}
        classes={
          Object {
            "defaultColors": "HvChart-defaultColors",
            "gridColor": "HvChart-gridColor",
            "lineColor": "HvChart-lineColor",
            "paddingTop": "HvChart-paddingTop",
            "plotColor": "HvChart-plotColor",
            "root": "HvChart-root",
            "subtitle": "HvChart-subtitle",
            "titleContainer": "HvChart-titleContainer",
            "vizText": "HvChart-vizText",
          }
        }
        data={
          Array [
            Object {
              "hoverinfo": "none",
              "name": "Sales Target",
              "type": "bar",
              "x": Array [
                "January",
                "February",
                "March",
              ],
              "y": Array [
                2300,
                1000,
                8500,
              ],
            },
          ]
        }
        layout={
          Object {
            "bargap": 0.25,
            "bargroupgap": 0.25,
          }
        }
        subtitle="Sales performance (YTD)"
        title="Simple Vertical Bar Chart"
        tooltipType="multiple"
        xAxisTitle="Thousands of Dollars ($)"
        yAxisTitle="Axis description"
      >
        <div
          className="HvChart-root"
        >
          <div
            className="HvChart-titleContainer"
          >
            <WithStyles(ForwardRef)
              variant="mTitle"
            >
              <ForwardRef
                classes={
                  Object {
                    "3xlTitle": "HvTypography-3xlTitle",
                    "4xlTitle": "HvTypography-4xlTitle",
                    "5xlTitle": "HvTypography-5xlTitle",
                    "disabledButtonText": "HvTypography-disabledButtonText",
                    "highlightText": "HvTypography-highlightText",
                    "infoText": "HvTypography-infoText",
                    "inlineLink": "HvTypography-inlineLink",
                    "lTitle": "HvTypography-lTitle",
                    "labelText": "HvTypography-labelText",
                    "mTitle": "HvTypography-mTitle",
                    "normalText": "HvTypography-normalText",
                    "placeholderText": "HvTypography-placeholderText",
                    "root": "HvTypography-root",
                    "sLink": "HvTypography-sLink",
                    "sText": "HvTypography-sText",
                    "sTitle": "HvTypography-sTitle",
                    "selectedNavText": "HvTypography-selectedNavText",
                    "selectedText": "HvTypography-selectedText",
                    "vizText": "HvTypography-vizText",
                    "xlTitle": "HvTypography-xlTitle",
                    "xsTitle": "HvTypography-xsTitle",
                    "xxlTitle": "HvTypography-xxlTitle",
                    "xxsTitle": "HvTypography-xxsTitle",
                  }
                }
                variant="mTitle"
              >
                <h3
                  className="HvTypography-root HvTypography-mTitle"
                >
                  Simple Vertical Bar Chart
                </h3>
              </ForwardRef>
            </WithStyles(ForwardRef)>
            <div
              className="HvChart-subtitle"
            >
              <WithStyles(ForwardRef)
                variant="sText"
              >
                <ForwardRef
                  classes={
                    Object {
                      "3xlTitle": "HvTypography-3xlTitle",
                      "4xlTitle": "HvTypography-4xlTitle",
                      "5xlTitle": "HvTypography-5xlTitle",
                      "disabledButtonText": "HvTypography-disabledButtonText",
                      "highlightText": "HvTypography-highlightText",
                      "infoText": "HvTypography-infoText",
                      "inlineLink": "HvTypography-inlineLink",
                      "lTitle": "HvTypography-lTitle",
                      "labelText": "HvTypography-labelText",
                      "mTitle": "HvTypography-mTitle",
                      "normalText": "HvTypography-normalText",
                      "placeholderText": "HvTypography-placeholderText",
                      "root": "HvTypography-root",
                      "sLink": "HvTypography-sLink",
                      "sText": "HvTypography-sText",
                      "sTitle": "HvTypography-sTitle",
                      "selectedNavText": "HvTypography-selectedNavText",
                      "selectedText": "HvTypography-selectedText",
                      "vizText": "HvTypography-vizText",
                      "xlTitle": "HvTypography-xlTitle",
                      "xsTitle": "HvTypography-xsTitle",
                      "xxlTitle": "HvTypography-xxlTitle",
                      "xxsTitle": "HvTypography-xxsTitle",
                    }
                  }
                  variant="sText"
                >
                  <p
                    className="HvTypography-root HvTypography-sText"
                  >
                    Sales performance (YTD)
                  </p>
                </ForwardRef>
              </WithStyles(ForwardRef)>
            </div>
          </div>
          <div
            className="HvChart-paddingTop"
            onMouseMove={[Function]}
          >
            <WithStyles(Plot)
              afterPlot={[Function]}
              config={
                Object {
                  "displayModeBar": false,
                  "responsive": true,
                }
              }
              data={
                Array [
                  Object {
                    "hoverinfo": "none",
                    "name": "Sales Target",
                    "type": "bar",
                    "x": Array [
                      "January",
                      "February",
                      "March",
                    ],
                    "y": Array [
                      2300,
                      1000,
                      8500,
                    ],
                  },
                ]
              }
              layout={
                Object {
                  "autosize": true,
                  "bargap": 0.25,
                  "bargroupgap": 0.25,
                  "colorway": Array [
                    "#2DB3E0",
                    "#FF990D",
                    "#D31F6B",
                    "#FFD00A",
                    "#18C38A",
                    "#098CB2",
                    "#EA5B86",
                    "#9B469B",
                    "#F4763D",
                    "#419393",
                    "#58C9DD",
                    "#F8C189",
                    "#528BBF",
                    "#DE64BD",
                    "#73BF9C",
                    "#9672C1",
                    "#EB95AF",
                    "#646A98",
                    "#D88136",
                    "#309B59",
                  ],
                  "legend": Object {
                    "font": Object {
                      "color": "#414141",
                      "family": Array [
                        "'Open Sans'",
                        "Arial",
                        "Helvetica",
                        "sans-serif",
                      ],
                      "size": 10,
                    },
                    "orientation": "h",
                    "x": 1,
                    "xanchor": "right",
                    "y": 1.1,
                    "yanchor": "bottom",
                  },
                  "margin": Object {
                    "b": 50,
                    "l": 50,
                    "pad": 0,
                    "t": 10,
                  },
                  "paper_bgcolor": "#FFFFFF",
                  "plot_bgcolor": "#FFFFFF",
                  "xaxis": Object {
                    "automargin": true,
                    "autorange": true,
                    "fixedrange": true,
                    "gridcolor": "#DEDEDE",
                    "linecolor": "#BCBCBC",
                    "linewidth": 1,
                    "range": Array [
                      -0.5,
                      2.5,
                    ],
                    "showgrid": false,
                    "tickcolor": "#BCBCBC",
                    "tickfont": Object {
                      "color": "#414141",
                      "family": Array [
                        "'Open Sans'",
                        "Arial",
                        "Helvetica",
                        "sans-serif",
                      ],
                      "size": 10,
                    },
                    "ticklen": 2,
                    "title": Object {
                      "font": Object {
                        "color": "#414141",
                        "family": Array [
                          "'Open Sans'",
                          "Arial",
                          "Helvetica",
                          "sans-serif",
                        ],
                        "size": 10,
                      },
                      "text": "Thousands of Dollars ($)",
                    },
                    "type": "category",
                    "zeroline": false,
                  },
                  "yaxis": Object {
                    "automargin": true,
                    "autorange": true,
                    "fixedrange": true,
                    "gridcolor": "#DEDEDE",
                    "linecolor": "#BCBCBC",
                    "linewidth": 1,
                    "range": Array [
                      0,
                      8947.368421052632,
                    ],
                    "showgrid": true,
                    "tickcolor": "#BCBCBC",
                    "tickfont": Object {
                      "color": "#414141",
                      "family": Array [
                        "'Open Sans'",
                        "Arial",
                        "Helvetica",
                        "sans-serif",
                      ],
                      "size": 10,
                    },
                    "ticklen": 2,
                    "title": Object {
                      "font": Object {
                        "color": "#414141",
                        "family": Array [
                          "'Open Sans'",
                          "Arial",
                          "Helvetica",
                          "sans-serif",
                        ],
                        "size": 10,
                      },
                      "text": "Axis description",
                    },
                    "type": "linear",
                    "zeroline": false,
                  },
                }
              }
              onHover={[Function]}
              onUnHover={[Function]}
              title="Simple Vertical Bar Chart"
            >
              <Plot
                afterPlot={[Function]}
                classes={Object {}}
                config={
                  Object {
                    "displayModeBar": false,
                    "responsive": true,
                  }
                }
                data={
                  Array [
                    Object {
                      "hoverinfo": "none",
                      "name": "Sales Target",
                      "type": "bar",
                      "x": Array [
                        "January",
                        "February",
                        "March",
                      ],
                      "y": Array [
                        2300,
                        1000,
                        8500,
                      ],
                    },
                  ]
                }
                layout={
                  Object {
                    "autosize": true,
                    "bargap": 0.25,
                    "bargroupgap": 0.25,
                    "colorway": Array [
                      "#2DB3E0",
                      "#FF990D",
                      "#D31F6B",
                      "#FFD00A",
                      "#18C38A",
                      "#098CB2",
                      "#EA5B86",
                      "#9B469B",
                      "#F4763D",
                      "#419393",
                      "#58C9DD",
                      "#F8C189",
                      "#528BBF",
                      "#DE64BD",
                      "#73BF9C",
                      "#9672C1",
                      "#EB95AF",
                      "#646A98",
                      "#D88136",
                      "#309B59",
                    ],
                    "legend": Object {
                      "font": Object {
                        "color": "#414141",
                        "family": Array [
                          "'Open Sans'",
                          "Arial",
                          "Helvetica",
                          "sans-serif",
                        ],
                        "size": 10,
                      },
                      "orientation": "h",
                      "x": 1,
                      "xanchor": "right",
                      "y": 1.1,
                      "yanchor": "bottom",
                    },
                    "margin": Object {
                      "b": 50,
                      "l": 50,
                      "pad": 0,
                      "t": 10,
                    },
                    "paper_bgcolor": "#FFFFFF",
                    "plot_bgcolor": "#FFFFFF",
                    "xaxis": Object {
                      "automargin": true,
                      "autorange": true,
                      "fixedrange": true,
                      "gridcolor": "#DEDEDE",
                      "linecolor": "#BCBCBC",
                      "linewidth": 1,
                      "range": Array [
                        -0.5,
                        2.5,
                      ],
                      "showgrid": false,
                      "tickcolor": "#BCBCBC",
                      "tickfont": Object {
                        "color": "#414141",
                        "family": Array [
                          "'Open Sans'",
                          "Arial",
                          "Helvetica",
                          "sans-serif",
                        ],
                        "size": 10,
                      },
                      "ticklen": 2,
                      "title": Object {
                        "font": Object {
                          "color": "#414141",
                          "family": Array [
                            "'Open Sans'",
                            "Arial",
                            "Helvetica",
                            "sans-serif",
                          ],
                          "size": 10,
                        },
                        "text": "Thousands of Dollars ($)",
                      },
                      "type": "category",
                      "zeroline": false,
                    },
                    "yaxis": Object {
                      "automargin": true,
                      "autorange": true,
                      "fixedrange": true,
                      "gridcolor": "#DEDEDE",
                      "linecolor": "#BCBCBC",
                      "linewidth": 1,
                      "range": Array [
                        0,
                        8947.368421052632,
                      ],
                      "showgrid": true,
                      "tickcolor": "#BCBCBC",
                      "tickfont": Object {
                        "color": "#414141",
                        "family": Array [
                          "'Open Sans'",
                          "Arial",
                          "Helvetica",
                          "sans-serif",
                        ],
                        "size": 10,
                      },
                      "ticklen": 2,
                      "title": Object {
                        "font": Object {
                          "color": "#414141",
                          "family": Array [
                            "'Open Sans'",
                            "Arial",
                            "Helvetica",
                            "sans-serif",
                          ],
                          "size": 10,
                        },
                        "text": "Axis description",
                      },
                      "type": "linear",
                      "zeroline": false,
                    },
                  }
                }
                onHover={[Function]}
                onUnHover={[Function]}
                title="Simple Vertical Bar Chart"
              >
                <PlotlyComponent
                  classes={Object {}}
                  config={
                    Object {
                      "displayModeBar": false,
                      "responsive": true,
                    }
                  }
                  data={
                    Array [
                      Object {
                        "hoverinfo": "none",
                        "name": "Sales Target",
                        "type": "bar",
                        "x": Array [
                          "January",
                          "February",
                          "March",
                        ],
                        "y": Array [
                          2300,
                          1000,
                          8500,
                        ],
                      },
                    ]
                  }
                  debug={false}
                  layout={
                    Object {
                      "autosize": true,
                      "bargap": 0.25,
                      "bargroupgap": 0.25,
                      "colorway": Array [
                        "#2DB3E0",
                        "#FF990D",
                        "#D31F6B",
                        "#FFD00A",
                        "#18C38A",
                        "#098CB2",
                        "#EA5B86",
                        "#9B469B",
                        "#F4763D",
                        "#419393",
                        "#58C9DD",
                        "#F8C189",
                        "#528BBF",
                        "#DE64BD",
                        "#73BF9C",
                        "#9672C1",
                        "#EB95AF",
                        "#646A98",
                        "#D88136",
                        "#309B59",
                      ],
                      "legend": Object {
                        "font": Object {
                          "color": "#414141",
                          "family": Array [
                            "'Open Sans'",
                            "Arial",
                            "Helvetica",
                            "sans-serif",
                          ],
                          "size": 10,
                        },
                        "orientation": "h",
                        "x": 1,
                        "xanchor": "right",
                        "y": 1.1,
                        "yanchor": "bottom",
                      },
                      "margin": Object {
                        "b": 50,
                        "l": 50,
                        "pad": 0,
                        "t": 10,
                      },
                      "paper_bgcolor": "#FFFFFF",
                      "plot_bgcolor": "#FFFFFF",
                      "xaxis": Object {
                        "automargin": true,
                        "autorange": true,
                        "fixedrange": true,
                        "gridcolor": "#DEDEDE",
                        "linecolor": "#BCBCBC",
                        "linewidth": 1,
                        "range": Array [
                          -0.5,
                          2.5,
                        ],
                        "showgrid": false,
                        "tickcolor": "#BCBCBC",
                        "tickfont": Object {
                          "color": "#414141",
                          "family": Array [
                            "'Open Sans'",
                            "Arial",
                            "Helvetica",
                            "sans-serif",
                          ],
                          "size": 10,
                        },
                        "ticklen": 2,
                        "title": Object {
                          "font": Object {
                            "color": "#414141",
                            "family": Array [
                              "'Open Sans'",
                              "Arial",
                              "Helvetica",
                              "sans-serif",
                            ],
                            "size": 10,
                          },
                          "text": "Thousands of Dollars ($)",
                        },
                        "type": "category",
                        "zeroline": false,
                      },
                      "yaxis": Object {
                        "automargin": true,
                        "autorange": true,
                        "fixedrange": true,
                        "gridcolor": "#DEDEDE",
                        "linecolor": "#BCBCBC",
                        "linewidth": 1,
                        "range": Array [
                          0,
                          8947.368421052632,
                        ],
                        "showgrid": true,
                        "tickcolor": "#BCBCBC",
                        "tickfont": Object {
                          "color": "#414141",
                          "family": Array [
                            "'Open Sans'",
                            "Arial",
                            "Helvetica",
                            "sans-serif",
                          ],
                          "size": 10,
                        },
                        "ticklen": 2,
                        "title": Object {
                          "font": Object {
                            "color": "#414141",
                            "family": Array [
                              "'Open Sans'",
                              "Arial",
                              "Helvetica",
                              "sans-serif",
                            ],
                            "size": 10,
                          },
                          "text": "Axis description",
                        },
                        "type": "linear",
                        "zeroline": false,
                      },
                    }
                  }
                  onAfterPlot={[Function]}
                  onHover={[Function]}
                  onUnhover={[Function]}
                  revision={0}
                  style={
                    Object {
                      "position": "relative",
                    }
                  }
                  title="Simple Vertical Bar Chart"
                  useResizeHandler={false}
                >
                  <div
                    style={
                      Object {
                        "position": "relative",
                      }
                    }
                  />
                </PlotlyComponent>
              </Plot>
            </WithStyles(Plot)>
          </div>
        </div>
      </Chart>
    </WithStyles(Chart)>
  </Barchart>
</WithStyles(Barchart)>
`;
